modules/mod_ls.c

```diff
@@ -205,9 +205,8 @@ static void push_cwd(char *_cwd, unsigned char *symhold) {
   if (!_cwd)
     _cwd = cwd;
 
-  *symhold = show_symlinks_hold;
   sstrncpy(_cwd, pr_fs_getcwd(), PR_TUNABLE_PATH_MAX + 1);
-  *symhold = list_show_symlinks;
+  *symhold = show_symlinks_hold = list_show_symlinks;
```

Comment:
This is the key fix: properly stash the current `list_host_symlinks` value in the `show_symlinks_hold` variable.  This has not been being done -- and thus broken -- for quite a while.